components/brave_wallet/browser/internal/orchard_sync_state.cc

```diff
@@ -0,0 +1,161 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/internal/orchard_sync_state.h"
+
+#include <utility>
+
+#include "base/check_is_test.h"
+#include "base/containers/extend.h"
+#include "base/types/expected_macros.h"
+#include "brave/components/brave_wallet/browser/internal/orchard_storage/orchard_types.h"
+#include "brave/components/brave_wallet/browser/zcash/rust/orchard_shard_tree.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+
+namespace brave_wallet {
+
+OrchardSyncState::OrchardSyncState(const base::FilePath& path_to_database)
+    : storage_(OrchardStorage(path_to_database)) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
```

Comment:
This is always going to be the case because the sequence checker is initialized on this thread. The only time the constructor and the rest of the code can run on a different sequence is if you call `DETACH_FROM_SEQUENCE`

Reply:
Done